src/app-layer-smtp.c

```diff
@@ -985,7 +985,9 @@ static int SMTPParseCommandBDAT(SMTPState *state, const SMTPLine *line)
     }
     memcpy(strbuf, line->buf + i, len);
     strbuf[len] = '\0';
-    if (StringParseUint32(&state->bdat_chunk_len, 10, 0, strbuf) < 0) {
+    char **endptr = NULL;
+    state->bdat_chunk_len = strtoul((const char *) strbuf, (char **) &endptr, 10);
```

Comment:
```
NOTES
       Since  strtoul() can legitimately return 0 or ULONG_MAX (ULLONG_MAX for strtoull()) on both success and failure, the calling pro‐
       gram should set errno to 0 before the call, and then determine if an error occurred by checking whether errno has a nonzero value
       after the call.
```
errno should be set to 0 before the call ^

Reply:
also from the manpage, this seems like it may be undesirable:
`Negative values are considered valid input and are silently converted to the equivalent unsigned long value.`

Reply:
@jlucovsky ^

Reply:
The current PR no longer uses `strtoul` -- see #11630 -- and has switched to `ByteExtractStringUint32`.  Values should be range-checked; there are existing `StringParse*RangeCheck` functions.